activerecord/lib/active_record/fixtures.rb

```diff
@@ -890,38 +902,44 @@ def set_fixture_class(class_names = {})
         self.fixture_class_names = fixture_class_names.merge(class_names.stringify_keys)
       end
 
-      def fixtures(*fixture_set_names)
-        if fixture_set_names.first == :all
-          fixture_set_names = Dir["#{fixture_path}/{**,*}/*.{yml}"].uniq
-          fixture_set_names.map! { |f| f[(fixture_path.to_s.size + 1)..-5] }
-        else
-          fixture_set_names = fixture_set_names.flatten.map(&:to_s)
+      def fixtures(*fixture_set_names, database: :all)
+        parent_fixture_table_names = fixture_table_names
+        self.fixture_table_names = Hash.new { |h, k| h[k] = [] }
+        configs = ActiveRecord::Base.configurations.configs_for(env_name: env)
+        configs.select! { |db_config| db_config.spec_name == database.to_s } unless database == :all
```

Comment:
We can simplify this a bit by leaning on `configs_for` for the spec name.

If you change the code like this:

```
configs = ActiveRecord::Base.configurations.configs_for(env_name: env, spec_name: database.to_s)
```

That will return an array of configs if `spec_name` is nil, or a single config if it's present so then we don't need to do the `select!`.